hts.c

```diff
@@ -1394,6 +1395,25 @@ int hts_set_fai_filename(htsFile *fp, const char *fn_aux)
     return 0;
 }
 
+hFILE *hts_open_tmpfile(const char *fname, const char *mode, kstring_t *tmpname)
+{
+    int pid = (int) getpid();
+    unsigned ptr = (uintptr_t) tmpname;
+    int n = 0;
+    hFILE *fp;
+
+    do {
+        // Attempt to further uniquify the temporary filename
+        unsigned t = ((unsigned) time(NULL)) ^ ((unsigned) clock()) ^ ptr;
+        n++;
+
+        ksprintf(ks_clear(tmpname), "%s.tmp_%d_%d_%u", fname, pid, n, t);
```

Comment:
A check on `ksprintf()` would be useful here, to prevent the possibility of passing NULL to `hopen()`.